src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs

```diff
@@ -475,6 +475,7 @@ private protected virtual void TrainCore(IChannel ch, RoleMappedData data)
                     e => e.SetProgress(0, exCount, totalCount));
                 while (cursor.MoveNext())
                 {
+                    Host.CheckAlive();
```

Comment:
>  Host.CheckAlive(); [](start = 18, length = 20)

is it too much to do it in every row fetch? would it be enough to do it every 10 cursor moves, or some other number > 1. 
(idk if there are any best practiced on how to determine the frequency of checks , from maybe the CancellationToken implementations)

Reply:
And how is that any more efficient than what we have now? You will end up executing an if condition on every row fetch ... based on my analysis of the current solution this doesn’t add any significant overhead.

Cancellation token works differently. You register a callback with it and when a signal is sent it invokes the callback and you do work to gracefully shutdown a process. Our plan is to implement cancellation token post 1.0.

Reply:
We spoke offline. I think this is the best we can do until we get cancellation tokens into the mix. `CheckAlive` only checks a `bool` property, so it's probably faster than checking to see if it's the 10th iteration or not.

Reply:
To add some late flavor to this, the branch predictor should be slightly better at the (almost perfectly) constant `bool` property, than the return value from `iteration % 10` (including the hidden division operation..could check every 8 as the compiler should optimize to a bitwise `AND`).

That said, there's the overhead of the `CheckAlive()` function call which maybe greater if not inlined.